### DISCUSSION TITLE
Return to Bookmark After Saving an Update

### DISCUSSION THREAD

**Discussion by paesjr** [directus/directus#23978] [Feature Requests]
### Summary

When updating a record accessed via a bookmark, Directus currently redirects users back to the main folder or collection view after saving. This behavior interrupts the workflow, as it forces users to re-navigate to the specific bookmark they were originally working within. 

### Motivation

It would be highly beneficial to return users to the exact bookmark view they came from after saving an update, reducing  repetitive steps, making Directus more intuitive and efficient.

### Detailed Design

Detailed Design
This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with Directus to understand, and for somebody familiar with the implementation to implement. This should get into specifics and corner-cases, and include examples of how the feature is used. Any new terminology should be defined here.

### Requirements List

Must Have
These are the basics to get this feature working right.

Bookmark Tracking:
Directus needs to know if you opened a record from a bookmark so it can remember that specific view or filter.

Redirect After Save:
Once you hit save, Directus should send you back to that bookmark view instead of dropping you at the main folder.

Toggle in Settings:
Include a simple switch in settings to turn this feature on or off, so users can choose the behavior they prefer.

Error Handling:
Have a backup plan if the bookmark’s unavailable (like if it got deleted). In that case, just send users to the default view.

Should Have
These make the feature even better but aren’t essential right away.

Bookmark in the URL:
Having a unique bookmark ID or link in the URL will make it easier to track where to go back to.

Smooth Transition:
Make the redirect look smooth—no awkward page reloads, so it feels like a seamless flow.

Save Confirmation:
A little pop-up or toast message after save that says "Saved! Returning to bookmark" could be helpful to reassure users.

Could Have
These extras would be nice but aren’t critical.

Bookmark History:
Track recent bookmarks in a session so users can go back and forth easily. Think of it like a “back to previous bookmark” button.

Bookmark-Level Preferences:
Let users set a custom redirect for each bookmark if they want, so each one behaves the way they like.

Keep Track After Logout:
If users log out or close their session, maybe remember the last bookmark context so they can pick up where they left off.

Won’t Have (for now)
Stuff that’s cool to think about but isn’t part of the plan right now.

Custom Layouts on Return:
Not looking to include options to customize the layout of each bookmark view—could be considered later if users want more control.

Integration with Flows:
Complex automations tied to specific bookmarks (like triggering actions based on the bookmark view) are also out for now, just to keep things simple.

### Drawbacks

Dev Complexity:
Tracking and redirecting back to bookmarks adds extra complexity to Directus’s navigation system. This means more code to write, test, and maintain, especially for scenarios with complex filters or nested views.

### Alternatives

Alternatives
Use Browser Tabs:
Users can open records in new tabs to avoid losing their place. Simple, but a bit clunky.

Add a “Return to Bookmark” Link:
Show a link after saving that takes users back to the bookmark if they want. Easier to implement and keeps user control.

Custom Redirect in Flows:
Power users could set up custom redirects using a Flow on save. It’s more advanced but keeps this as an option without Directus adding a feature.

Impact of Not Doing This
Mild Inconvenience: Users who heavily use bookmarks might find it a bit slower if they have to navigate back manually.
Keeps Directus Simple: No extra settings or navigation complexity, so it stays straightforward.
Focus on Bigger Improvements: Skipping this frees up time for more widely useful updates.

### Adoption Strategy

Adoption Strategy
If we do this, here’s the plan:

No Breakage:
It won’t break anything, just adds an option in settings. Existing projects won’t have to change anything.

Easy On/Off Toggle:
Add a setting to turn it on or off, so users can try it without committing.

Simple Docs Update:
Just a short guide or note in the docs to help users get started.

Community Feedback:
Launch it, see how users like it, and adjust based on feedback. If it’s popular, great; if not, it’s easy to phase out.

**Comment by Xinnerpeace** [comment]
This is a simple change, but with a very high trigger frequency.

**Comment by ertanar** [comment]
Isn't this a duplicate of https://github.com/directus/directus/issues/4025 ?
